### PR TITLE
Adds YT Web case study - Part 1

### PR DESCRIPTION
**Pull request by addyosmani**
This draft has been iterated on as part of https://github.com/GoogleChrome/developer.chrome.com/pull/4049 and we are now moving it over to web.dev with a view to publishing here instead.

Draft previously reviewed by Phil Walton, Lauren, Harleen, Jeremy and reps from both YT and Chrome. 

Changes:
- Adds co-author to our authors
- Adds the case study

Before publishing:
- Update the publish date (likely to the 20th)

### CODE REVIEW

File: src/site/content/en/blog/better-youtube-web-part1/index.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+and limited network bandwidth, ensuring a fast and seamless experience is a
+critical goal.
+
+To provide an inclusive experience for all users, YouTube set out to improve
+performance metrics such as Core Web Vitals through lazy-loading and code
+modernization.
+
+### Improving Core Web Vitals
+
+To identify areas of improvement, the YouTube team used the Chrome User
```
Comment (philipwalton) [inline src/site/content/en/blog/better-youtube-web-part1/index.md:55]: Nit: for consistency with the other links, I'd apply the link to the whole name:

```md
[Chrome User Experience Report (CrUX)](https://developer.chrome.com/docs/crux/)
```

File: src/site/content/en/blog/better-youtube-web-part1/index.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+{% Img src="image/1L2RBhCLSnXjCnSlevaDjy3vba73/wGjBH4sMWoZEbRon6gBS.png", alt="The black thumbnail was deployed in production for all mobile web users July 2021 showing marked improvement in FCP and LCP, as seen in the above RUM analysis.", width="800", height="408" %}
+  <figcaption>
+    Black thumbnail was deployed in production for all mobile web users July 2021 showing marked improvement in FCP and LCP, as seen in the above RUM analysis.
+  </figcaption>
+</figure>
+
+{% Aside%}
+While bringing these optimizations to all platforms, YouTube also took advantage of the new [Priority Hints](/priority-hints/) `fetchpriority` attribute, which we use with `<link rel=preload>` to prioritize discovering and loading the poster image early:
+
+```html
```
Comment (philipwalton) [inline src/site/content/en/blog/better-youtube-web-part1/index.md:106]: This is not rendering properly for some reason? It looks like the markdown processor isn't registering this as a code block?

Not sure what the best solution is, but a short term solution may be to just manually add the HTML? WDYT?

```
<pre class="language-html"><code class="language-html"><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>link</span> <span class="token attr-name">as</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">"</span>image<span class="token punctuation">"</span></span> <span class="token attr-name">rel</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">"</span>preload<span class="token punctuation">"</span></span> <span class="token attr-name">href</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">"</span>poster.jpg<span class="token punctuation">"</span></span> <span class="token attr-name">fetchpriority</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">"</span>high<span class="token punctuation">"</span></span><span class="token punctuation">&gt;</span></span></code></pre>
```

File: src/site/content/en/blog/better-youtube-web-part1/index.md [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+authors:
+  - addyosmani
+  - sriramkrishnan
+hero: 'image/1L2RBhCLSnXjCnSlevaDjy3vba73/CPrVgVl7t1MuwWpXzvCE.jpg'
+alt: >
+  A hero image of the YouTube mobile web loading experience.
+tags:
+  - blog # blog is a required tag for the article to show up in the blog.
+  - case-study
+  - performance
```
[the remark this answers is not in the document]
Comment (addyosmani) [reply src/site/content/en/blog/better-youtube-web-part1/index.md:18]: Sure!

File: src/site/content/en/blog/better-youtube-web-part1/index.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+  - web-vitals
+---
+
+The Chrome team often talks about "building a *better* web", but what does that
+mean? Web experiences should be [fast](/why-speed-matters/),
+[accessible](/accessibility/), and leverage [device
+capabilities](/reliable/) in the moment when users need it most.
+Dogfooding is part of Google's culture, so the Chrome team has partnered with
```
Comment (philipwalton) [inline src/site/content/en/blog/better-youtube-web-part1/index.md:26]: (Optional) Maybe link to this for the word "Dogfooding":
https://en.wikipedia.org/wiki/Eating_your_own_dog_food

File: src/site/content/en/blog/better-youtube-web-part1/index.md [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+responds to user interactions such as search, player control, likes, shares, and
+many others.
+
+Growing developing markets, such as Brazil, India, and Indonesia are important
+for YouTube mobile web. Because many users in these regions have slower devices
+and limited network bandwidth, ensuring a fast and seamless experience is a
+critical goal.
+
+To provide an inclusive experience for all users, YouTube set out to improve
+performance metrics such as Core Web Vitals through lazy-loading and code
```
Comment (philipwalton) [inline src/site/content/en/blog/better-youtube-web-part1/index.md:51]: I know it's linked above (in the figcaption), but I'd like "Core Web Vitals" here as well.

File: src/site/content/en/blog/better-youtube-web-part1/index.md [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+
+## Conclusion
+
+As a result of YouTube's investment in performance, watch pages load much faster with 76% of YouTube's mobile website URLs now passing Core Web Vitals metric thresholds in the field. On desktop, lab LCP for the watch page was reduced from approximately 4.6 seconds to 1.6 seconds. Interaction and rendering performance of the site, especially on the YouTube video player, are seeing up to 75% less time spent in JavaScript execution than before.
+
+{% Aside 'success' %}
+76% of YouTube's mobile web URLs now pass the Core Web Vitals metrics, with improvements to business metrics like watch time as a result.
+{% endAside %}
+
+Improvements to the performance of YouTube web over the last year have also improved business metrics, including watch time and daily active users. We look forward to exploring how more of these performance optimizations can be applied to the YouTube desktop site.
```
Comment (philipwalton) [inline src/site/content/en/blog/better-youtube-web-part1/index.md:176]: I think this last sentence can be dropped (or reworded to talk about general future optimizations) now that the post doesn't focus on the specific mobile and desktop changes. Not sure if this is true, but maybe something inspirational like:

"Based on the success of these efforts, we plan to continue exploring even more ways to optimize in the future."

**Review by philipwalton — APPROVED** [review]
LGTM (modulo a few comments)

File: src/site/content/en/blog/better-youtube-web-part1/index.md [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+tags:
+  - blog # blog is a required tag for the article to show up in the blog.
+  - case-study
+  - performance
+  - web-vitals
+---
+
+The Chrome team often talks about "building a *better* web", but what does that
+mean? Web experiences should be [fast](/why-speed-matters/),
+[accessible](/accessibility/), and leverage [device
```
Comment (rachelandrew) [inline src/site/content/en/blog/better-youtube-web-part1/index.md:24]:
```suggestion
[accessible](/accessibility/), and use [device
```

Comment (rachelandrew) [reply src/site/content/en/blog/better-youtube-web-part1/index.md:24]: see note here https://developers.google.com/style/word-list#letter-l

File: src/site/content/en/blog/better-youtube-web-part1/index.md [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+
+To identify areas of improvement, the YouTube team used the [Chrome User Experience Report (CrUX)](https://developer.chrome.com/docs/crux/) to see how real users were experiencing video watch and search result pages on
+mobile in [the
+field](/lab-and-field-data-differences/#field-data). They
+realized their Core Web Vitals metrics had a lot of room for improvement, with
+their [Largest Contentful Paint (LCP)](/lcp/) metric clocking in
+at 4-6 seconds in some cases. This was substantially higher than their target of
+2.5 seconds.
+
+{% Img src="image/1L2RBhCLSnXjCnSlevaDjy3vba73/uy6hO7M60rRw3bTfXTCH.png", alt="Charts of FCP and LCP showing YouTube Watch page pass rates as well as the YouTube origin", width="800", height="285" %}
```
Comment (rachelandrew) [inline src/site/content/en/blog/better-youtube-web-part1/index.md:64]:
```suggestion
{% Img src="image/1L2RBhCLSnXjCnSlevaDjy3vba73/uy6hO7M60rRw3bTfXTCH.png", alt="Charts of FCP and LCP showing YouTube Watch page pass rates as well as the YouTube origin.", width="800", height="285" %}
```

File: src/site/content/en/blog/better-youtube-web-part1/index.md [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+resulting in two big discoveries.
+
+1. The first discovery was that they could improve performance by moving the HTML for the Video Player above the script that makes the Video Player interactive. Lab tests indicated that this could improve both FCP and LCP from 4.4 seconds to 1.1 seconds.
+
+1. The second discovery was that LCP only [considers](/lcp/#what-elements-are-considered) `<video>` element poster images and not frames from the video stream itself. YouTube has traditionally optimized for the fastest time until the video starts playing, so to improve LCP the team started also optimizing how quickly they could deliver their poster image. They experimented with a few variations of poster images and picked the one that scored the best in user testing. As a result of this work, both FCP and LCP showed marked improvement, with field LCP improving from 4.6 seconds to 2.0 seconds.
+  
+<figure>
+{% Img src="image/1L2RBhCLSnXjCnSlevaDjy3vba73/wJKAgsXRqrexp9yDJ4T1.jpg", alt="Watch Page LCP Experiment for mobile web showing control, experiment A (image thumbnail) and experiment B (black thumbnail)", width="800", height="514" %}
```
Comment (rachelandrew) [inline src/site/content/en/blog/better-youtube-web-part1/index.md:88]:
```suggestion
{% Img src="image/1L2RBhCLSnXjCnSlevaDjy3vba73/wJKAgsXRqrexp9yDJ4T1.jpg", alt="Watch Page LCP Experiment for mobile web showing control, experiment A (an image thumbnail) and experiment B (a black thumbnail).", width="800", height="514" %}
```

File: src/site/content/en/blog/better-youtube-web-part1/index.md [hunk 10]
```diff
@@ -0,0 +1,180 @@
...
+
+Once this change lands in Chrome, the YouTube team is excited to continue their work optimizing for LCP. And the updated version of the metric will mean these optimizations will have a more direct impact on real-user experiences.
+
+### Modularization with lazy loading
+
+YouTube pages contained many modules that were eagerly loaded. To optimize how
+50+ components were rendered, the team built a component to JS module map that
+would tell the client which modules to load. By marking components as lazy, the
+JS modules would load later, reducing the initial load time of the page and the
+amount of unused Javascript sent to the client.
```
Comment (rachelandrew) [inline src/site/content/en/blog/better-youtube-web-part1/index.md:119]:
```suggestion
YouTube pages contained many modules that were eagerly loaded. To optimize how
more than 50 components were rendered, the team built a component to JavaScript module map that
would tell the client which modules to load. By marking components as lazy, the
JavaScript modules would load later, reducing the initial load time of the page and the
amount of unused Javascript sent to the client.
```

File: src/site/content/en/blog/better-youtube-web-part1/index.md [hunk 11]
```diff
@@ -0,0 +1,180 @@
...
+### State-management across components
+
+YouTube was experiencing performance issues due to its player controls,
+especially on older devices. Code analysis showed that the player, which allows
+users to control features such as playback speed and progress, had become
+over-componentized over time. 
+
+
+<figure>
+{% Img src="image/1L2RBhCLSnXjCnSlevaDjy3vba73/qh9N7ZdkxylN1vYOOdUO.png", alt="YouTube player and controls visualized", width="800", height="300" %}
```
Comment (rachelandrew) [inline src/site/content/en/blog/better-youtube-web-part1/index.md:139]:
```suggestion
{% Img src="image/1L2RBhCLSnXjCnSlevaDjy3vba73/qh9N7ZdkxylN1vYOOdUO.png", alt="YouTube player and controls visualized.", width="800", height="300" %}
```

File: src/site/content/en/blog/better-youtube-web-part1/index.md [hunk 12]
```diff
@@ -0,0 +1,180 @@
...
+  <figcaption>
+    The YouTube video player allows users to control playback speed, follow progress, skip sections, and others. When a user taps on a particular control, the change in the state must be communicated to other controls e.g. a user tap on the progress bar must be shared to controls such as play-head, subtitles etc.
+  </figcaption>
+</figure>
+
+Each progress bar touch-move event triggered two
+extra style recalculations and took 21.17 ms during performance test runs in the lab. As new controls were added over time, the pattern of decentralized control would often cause circular dependencies and memory leaks, negatively impacting watch page performance.
+
+<figure>
+{% Img src="image/1L2RBhCLSnXjCnSlevaDjy3vba73/vCTMilflEoDnkw1p8EuU.png", alt="21.17ms event shown on the Performance timeline", width="800", height="202" %}
```
Comment (rachelandrew) [inline src/site/content/en/blog/better-youtube-web-part1/index.md:149]:
```suggestion
{% Img src="image/1L2RBhCLSnXjCnSlevaDjy3vba73/vCTMilflEoDnkw1p8EuU.png", alt="21.17 ms event shown on the Performance timeline.", width="800", height="202" %}
```

File: src/site/content/en/blog/better-youtube-web-part1/index.md [hunk 13]
```diff
@@ -0,0 +1,180 @@
...
+  <figcaption>
+    Chrome DevTools with a 4x CPU slow-down performance run.
```
Comment (rachelandrew) [inline src/site/content/en/blog/better-youtube-web-part1/index.md:151]:
```suggestion
    Chrome DevTools with a 4 times CPU slow-down performance run. 
```

Comment (rachelandrew) [reply src/site/content/en/blog/better-youtube-web-part1/index.md:151]: https://developers.google.com/style/abbreviations#spelling-out

File: src/site/content/en/blog/better-youtube-web-part1/index.md [hunk 14]
```diff
@@ -0,0 +1,180 @@
...
+</figure>
+
+To fix the issues due to decentralized control, the team updated the player UI
+to synchronize all updates, essentially refactoring the player to one top-level
+component that would pass down data to its children. This ensured only one UI
+update (render) cycle for any state change, eliminating chained updates. The new
+player progress bar touch-move event has no style recalculations during its
+JavaScript execution and now only requires 25% the time of the old player.
+
+{% Img src="image/1L2RBhCLSnXjCnSlevaDjy3vba73/E1EbOYuT2q7mDqfmR1zz.png", alt="Reduced time in events shown on the performance timeline", width="800", height="90" %}
```
Comment (rachelandrew) [inline src/site/content/en/blog/better-youtube-web-part1/index.md:162]:
```suggestion
{% Img src="image/1L2RBhCLSnXjCnSlevaDjy3vba73/E1EbOYuT2q7mDqfmR1zz.png", alt="Reduced time in events shown on the performance timeline.", width="800", height="90" %}
```

File: src/site/content/en/blog/better-youtube-web-part1/index.md [hunk 15]
```diff
@@ -0,0 +1,180 @@
...
+    The YouTube for mobile web Watch page passing the <a href="/vitals">Core Web Vitals</a> thresholds 
+  </figcaption>
+</figure>
+
+## Building a faster web
+
+Performance to YouTube is how fast videos and other content, such as
+recommendations and comments, load on web pages. It is also how quickly it
+responds to user interactions such as search, player control, likes, shares, and
+many others.
```
Comment (rachelandrew) [inline src/site/content/en/blog/better-youtube-web-part1/index.md:43]:
```suggestion
At YouTube, _performance_ relates to how fast videos and other content—such as
recommendations and comments—load on web pages. Performance is also measured by how quickly YouTube
responds to user interactions such as search, player control, likes, and shares.
```

Comment (rachelandrew) [reply src/site/content/en/blog/better-youtube-web-part1/index.md:43]: This was a confusing paragraph with two "it" referring to different things in the second sentence. I think this reads better.

File: src/site/content/en/blog/better-youtube-web-part1/index.md (hunk 4 above)
Comment (rachelandrew) [reply src/site/content/en/blog/better-youtube-web-part1/index.md:26]: yes please.

File: src/site/content/en/blog/better-youtube-web-part1/index.md (hunk 6 above)
Comment (addyosmani) [reply src/site/content/en/blog/better-youtube-web-part1/index.md:175]: Sounds good! I'll go ahead and drop this last line and replace with your suggested wording.

File: src/site/content/en/blog/better-youtube-web-part1/index.md (hunk 4 above)
Comment (addyosmani) [reply src/site/content/en/blog/better-youtube-web-part1/index.md]: SGTM

**Pull request merged**